server/eclipse-project/src/test/java/au/org/democracydevelopers/corla/csv/DominionCVRExportParserTests.java

```diff
@@ -0,0 +1,585 @@
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+See the GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along with
+raire-service. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+package au.org.democracydevelopers.corla.csv;
+
+import au.org.democracydevelopers.corla.model.ContestType;
+import au.org.democracydevelopers.corla.testUtils;
+
+import org.apache.log4j.LogManager;
+import org.apache.log4j.Logger;
+import org.testcontainers.containers.PostgreSQLContainer;
+import us.freeandfair.corla.csv.DominionCVRExportParser;
+import us.freeandfair.corla.model.*;
+import us.freeandfair.corla.persistence.Persistence;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.testng.annotations.*;
+
+import static au.org.democracydevelopers.corla.testUtils.*;
+import static org.testng.Assert.*;
+
+import static us.freeandfair.corla.query.CastVoteRecordQueries.getMatching;
+import static us.freeandfair.corla.query.ContestQueries.forCounties;
+import static us.freeandfair.corla.query.CountyQueries.fromString;
+
+/**
+ * Test a variety of IRV-related parsing issues for CVRs, including:
+ * - some basic small test cases that can be verified by eye,
+ * - a constructed test with invalid IRV ballots to check they are accepted and properly
+ *   interpreted.
+ * - a real example from Boulder '23, with a mix of IRV and plurality contests
+ * - some examples with invalid headers, to ensure they are rejected (though most of these are
+ *   tested in IRVHeadersParserTests.java and are not repeated here),
+ *   plus a bad plurality "Vote For= " with a non-integer.
+ * - an examples to test the broader class of Write In strings.
+ */
+public class DominionCVRExportParserTests {
+
+  /**
+   * Class-wide logger
+   */
+  private static final Logger LOGGER = LogManager.getLogger(DominionCVRExportParserTests.class);
+
+  /**
+   * Container for the mock-up database.
+   */
+  static PostgreSQLContainer<?> postgres
+      = new PostgreSQLContainer<>("postgres:15-alpine")
+      // None of these actually have to be the same as the real database (except its name), but this
+      // makes it easy to match the setup scripts.
+      .withDatabaseName("corla")
+      .withUsername("corlaadmin")
+      .withPassword("corlasecret")
+      .withInitScript("corlaInit.sql");
+
+  /**
+   * Error message to match.
+   */
+  private static final String badNumsRegexp = "Unexpected or uninterpretable numbers in header:.*";
+
+  /**
+   * Blank properties for submitting to the DominionCVRExportParser instance.
+   */
+  private static final Properties blank = new Properties();
+
```

Comment:
Should the "org.postgresql.Driver" and "org.hibernate.dialect.PostgreSQL9Dialect" be hardcoded here?

Reply:
No, probably not - see above.